tests/monitors/haproxy/haproxy_test.py

```diff
@@ -33,7 +33,23 @@ def test_haproxy_default_metrics_from_stats_page(version):
              url: http://{host}:8080/stats?stats;csv
            """
         ) as agent:
+            verify(agent, EXPECTED_DEFAULTS - EXPECTED_DEFAULTS_FROM_SOCKET, 10)
+            assert not has_log_message(agent.output.lower(), "error"), "error found in agent output!"
+
 
+@pytest.mark.parametrize("version", ["1.8"])
+def test_haproxy_default_metrics_from_stats_page_by_discovery_rule(version):
+    with run_service("haproxy", buildargs={"HAPROXY_VERSION": version}, name="haproxy", ports={8080: 8080}):
```

Comment:
In `test-services/haproxy/Dockerfile`, add a line that says `EXPOSE 8080` and then you shouldn't have to do the `ports={8080:8080}` thing, which is not very robust since you couldn't run multiple tests with the same port simultaneously.

Reply:
ok